nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -135,6 +136,14 @@ def __init__(self, cfg: DictConfig, trainer: Trainer, no_lm_init=True):
             "default_on_epoch": False,
         }
 
+        self.gc_interval = cfg.get('gc_interval', 0)
+        assert self.gc_interval >= 0, "gc_interval should be an integer value larger than or equal to 0."
+        # If gc_interval > 0, memory garbage collection is manually controlled.
+        # The automatic garbage collector sould be disabled before training starts.
+        if self.gc_interval > 0:
+            gc.disable()
+            self.global_batch_count = 0
```

Comment:
also, do we need to maintain our own attribute for this, can we use self.trainer.global_step?